File: src/utils/processor.py
import json
from bs4 import BeautifulSoup
import datetime
from models.article import Article
from fetcher import fetch

def extract(email_content):
    soup = BeautifulSoup(email_content, 'html.parser')

    title = soup.find('h1').text if soup.find('h1') else None
    image = soup.find('img')['src'] if soup.find('img') else None
    date = datetime.datetime.now().isoformat()
    content = str(soup)
    snippet = soup.find('p').text[:40] if soup.find('p') else None

    article = Article(title, image, date, content, snippet)

    with open('../articles.json', 'w') as json_file:
        json.dump(article.__dict__, json_file)

if __name__ == "__main__":
    email_content = fetch()
    if email_content:
        extract(email_content)

File: utils/processor.ts
import { parse } from 'node-html-parser';
import { fetch } from './fetcher';
import { Post } from './post';

export const globalState = {
    postGlobal: null as Post | null
};

async function extract() {
    const emailContent = await fetch();

    if (emailContent) {
        const root = parse(emailContent);

        const title = root.querySelector('h1')?.text;
        const image = root.querySelector('img')?.getAttribute('src');
        const date = new Date().toISOString();
        const content = root.toString();
        const snippet = root.querySelector('p')?.text.slice(0, 40);

        const words = content.split(' ').length;
        const readtime = Math.ceil(words / 200);

        globalState.postGlobal = new Post(title || '', image || '', date, content, snippet || '', readtime);
    }
}

export { extract };
